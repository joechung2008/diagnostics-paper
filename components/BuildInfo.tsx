import { Text, View } from "react-native";
import { useBuildInfoStyles } from "../hooks/useBuildInfoStyles";
import type { BuildInfoProps } from "../lib/types";

const BuildInfo = ({ buildVersion }: BuildInfoProps) => {
  const items = [
    {
      name: "Build Version",
      value: buildVersion,
    },
  ];

  const dynamicStyles = useBuildInfoStyles();

  return (
    <View style={dynamicStyles.container}>
      <View style={dynamicStyles.table}>
        <View style={dynamicStyles.header}>
          <Text style={dynamicStyles.headerText}>Name</Text>
          <Text style={dynamicStyles.headerText}>Value</Text>
        </View>
        {items.map((item) => (
          <View key={item.name} style={dynamicStyles.row}>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.name}
            </Text>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.value}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};

export default BuildInfo;
