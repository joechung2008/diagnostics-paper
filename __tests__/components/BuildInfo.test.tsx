import React from "react";
import { PaperProvider, MD3LightTheme } from "react-native-paper";
import { render } from "@testing-library/react-native";
import BuildInfo from "../../components/BuildInfo";

describe("BuildInfo", () => {
  it("renders correctly with build version", () => {
    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <BuildInfo buildVersion="1.0.0" />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
