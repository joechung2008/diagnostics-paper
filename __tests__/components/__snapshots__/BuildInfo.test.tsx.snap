// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildInfo renders correctly with build version 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        Build Version
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        1.0.0
      </Text>
    </View>
  </View>
</View>
`;
